### ISSUE TITLE
Port twisted.internet.endpoint's tcp endpoints to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6088]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6088|
|Type|enhancement|
|Created|2012-10-09 18:24:44Z|
|Branch|https://github.com/twisted/twisted/tree/endpoints-py3-6088|

`twisted.internet.endpoints` should be ported to Python 3, in a minimal fashion, speicifically TCP (necessary for [#6069](https://github.com/twisted/twisted/issues/6069)) to begin with.

<details><summary>Searchable metadata</summary>

```
trac-id__6088 6088
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_endpoints_py3_6088 branches/endpoints-py3-6088
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349807084000000 1349807084000000
changetime__1349972721000000 1349972721000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36030]) Branching to 'endpoints-py3-6088'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/endpoints-py3-6088 is running.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. twisted/internet/endpoints.py imports `_WrappingProtocol` but does not use it
  1. twisted/internet/_endpointspy3.py is missing `__future__` imports

I reviewed the moved code for differences and didn't find any of significance.  I would have been confident to skip that step given a note mentioning none of the code had changed (apart from `implements`/`implementer`).

Looks good apart from those two points, please merge when they're resolved.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Oh damn, I forgot to be snarky.

  3. I don't see any ported UDP code, did you forget to do that?

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_8"></a>|@itamarst set status to `closed`|
|-|-|

(In [36083]) Merge endpoints-py3-6088.

Author: itamar
Review: exarkun
Fixes: [#6088](https://github.com/twisted/twisted/issues/6088)

Port TCP endpoints to Python 3.
